blog/content/post/2024-04-12-odroid-m1-secure-boot.md

```diff
@@ -0,0 +1,649 @@
+export ROCKCHIP_TPL=<path/to/rkbin>/bin/rk35/rk3568_ddr_1560MHz_v1.21.bin
+```
+
+This time I used `gcc-aarch64-linux-gnu` cross-compiler from Debian package
+manager.
+
+### Signing U-Boot
+
+To sign U-Boot I used mkimage:
+
+```shell
+tools/mkimage -F -k . u-boot.itb
+FIT description: FIT image for U-Boot with bl31 (TF-A)
+Created:         Tue Apr 16 17:37:12 2024
+ Image 0 (u-boot)
+  Description:  U-Boot
+  Created:      Tue Apr 16 17:37:12 2024
+  Type:         Standalone Program
+  Compression:  uncompressed
+  Data Size:    854080 Bytes = 834.06 KiB = 0.81 MiB
+  Architecture: AArch64
+  Load Address: 0x00a00000
+  Entry Point:  0x00a00000
+  Hash algo:    sha256
+  Hash value:   c51061551a3183de5b1d2c7ec792d782af89530663ea9bd6a47c6e2e3340b99f
+(...)
+ Image 7 (fdt-1)
+  Description:  fdt-rk3568-odroid-m1
+  Created:      Tue Apr 16 17:37:12 2024
+  Type:         Flat Device Tree
+  Compression:  uncompressed
+  Data Size:    79264 Bytes = 77.41 KiB = 0.08 MiB
+  Architecture: Unknown Architecture
+  Hash algo:    sha256
+  Hash value:   d3a940c539c62eb5702efafe73a41fb4f4a25c00db556ba7e33d2f86f1f18643
+ Default Configuration: 'config-1'
+ Configuration 0 (config-1)
+  Description:  rk3568-odroid-m1.dtb
+  Kernel:       unavailable
+  Firmware:     atf-1
+  FDT:          fdt-1
+  Loadables:    u-boot
+                atf-2
+                atf-3
+                atf-4
+                atf-5
+                atf-6
+  Sign algo:    sha256,rsa2048:dev
+  Sign value:   17f41ebae64c17bac98ac03220048c0c87cf82ccd8a145ab828f766a07a0526a88d9cdf2699422b99b9ad6ba6981921d1ace9e938a8c94cd8c9f9c032a8422666fd076efb21e62f8311b5443245cd7aa78ecedce13f463a9dadec0635d9c63dd195d8bf948e2abd79f3b252aa71e34719bb9debd5bece717661d6128fe68c9b98dbb2f429abc00c2b7e5bee7cdda2fed1ac95c5dc378d69cc06448de68e3588acfc1055af96193866be383b3c48e7946dafe214d634a597aaae84496c1e53259078d34181de32fb7a9c902540bf891c4d31e6ba956a31b875123e90cb4430d093895b218f4aba6a361a0e8e7cf339e60c8eb38ad38480df171c082e1a578f692
+  Timestamp:    Tue Apr 16 17:43:41 2024
+Signature written to 'u-boot.itb', node '/configurations/config-1/signature'
+```
+
+### Signing idbloader
+
+To sign idbloader I used once again `rk_sign_tool` from rkbin repository.
+Keys used to sign idbloader need to be the same that we used with Rockchip
+U-Boot in previous stage.
+
+```shell
+tools/rk_sign_tool sb --idb ../u-boot/idbloader.img
+********sign_tool ver 1.4********
+IDB binary is ../u-boot/idbloader.img
+signing idbhead...
+failed to get key = sign_algo
+signing idbloader ok
+```
+
+### U-Boot Verification
+
+To check if SPL is signed correctly and that it correctly verifies U-Boot I have
+written idbloader.img file to `spl` partition created in
+[Writing key hash to OTP](#writing-key-hash-to-otp). I also flashed `u-boot.itb`
+file to `uboot` partition.
+
+After writing needed files and inserting SD card into ODROID i restarted
+platform while keeping recovery button pressed.
+
+```text
```

Comment:
@m-iwanicki do not you boot up until U-Boot shell? We could show that

Reply:
In [11a1594](https://github.com/3mdeb/news-and-ideas/pull/593/commits/11a15940ed80c91310292ac7654018debeb2a87f) I added

```
(...)
=>
```

Whole output would be too long